pkg/minishift/instancedata/instancedata.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instancedata
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+// Variable to be used anywhere
+var Data = new(InstanceData)
+
+type InstanceData struct {
+	FilePath string `json:"-"`
+	OcPath   string
+}
+
+func (m *InstanceData) Initialize(path string) {
+	m.FilePath = path
+}
+
+// Create instance-data.json with empty data or use existing data
+func (m *InstanceData) Create() error {
```

Comment:
Ok, 👍 for the instance methods. Since they are instance methods now, we might as well just cal them _Create_, _Write_, _Read_. Create does imo not even have to be public, but is called only when the instance is created. In fact there is a problem with the code now anways, since I can call _Create_ on an instance which does not have the path set.